src/app/pages/ValidatorDetailsPage/index.tsx

```diff
@@ -1,24 +1,63 @@
 import { FC } from 'react'
 import { useTranslation } from 'react-i18next'
+import Card from '@mui/material/Card'
+import CardContent from '@mui/material/CardContent'
+import Divider from '@mui/material/Divider'
 import { useScreenSize } from '../../hooks/useScreensize'
-import { Validator } from '../../../oasis-nexus/api'
+import { Validator, useGetConsensusValidatorsEntityId } from '../../../oasis-nexus/api'
 import { StyledDescriptionList } from '../../components/StyledDescriptionList'
 import { PageLayout } from '../../components/PageLayout'
 import { TextSkeleton } from '../../components/Skeleton'
 import { StatusIcon } from '../../components/StatusIcon'
 import { ValidatorImage } from '../../components/Validators/ValidatorImage'
 import { ValidatorCommission } from '../../components/Validators/ValidatorCommission'
 import { ValidatorCumulativeVoting } from '../../components/Validators/ValidatorCumulativeVoting'
+import { ValidatorTitleCard } from './ValidatorTitleCard'
+import { useRequiredScopeParam } from 'app/hooks/useScopeParam'
+import { useLoaderData } from 'react-router-dom'
+import { AddressLoaderData } from 'app/utils/route-utils'
+import { ValidatorSnapshot } from './ValidatorSnapshot'
 
 export const ValidatorDetailsPage: FC = () => {
-  return <PageLayout>{/* TODO: Implement validator details cards */}</PageLayout>
+  const { isMobile } = useScreenSize()
+  const scope = useRequiredScopeParam()
+  // TODO: currently API does not work with address query param. Wait for API update or switch to entity_id
```

Comment:
Yeah, that sucks. Also when looking at network change proposal votes, we get the address of the validators, and not the entity ID.

Reply:
in addition entity_id is using valid URL chars (like  / + ) that won't be escaped by encodeURIComponent (we would need custom function). Anyway, added a note for backend to change current query param.